src/v/archival/archival_policy.cc

```diff
@@ -241,18 +241,26 @@ static ss::future<> get_file_range(
           "Full segment upload {}, file size: {}",
           upl,
           fsize);
-        co_return;
+        co_return storage::convert_to_file_pos_outcome::success;
     }
     if (begin_inclusive != segment->offsets().base_offset) {
-        auto seek = co_await storage::convert_begin_offset_to_file_pos(
+        auto seek_result = co_await storage::convert_begin_offset_to_file_pos(
           begin_inclusive, segment, upl->base_timestamp, io_priority);
+        if (seek_result.error()) {
+            co_return seek_result.error();
+        }
```

Comment:
this looks like undefined behavior. did you mean to check if the result contained an error?